### PR TITLE
Validator

### LINKED ISSUE #779: update openfhe-lattice-estimator functionality
- update cpp scripts to take num_of_inputs as a configuration parameter; add validation checks
- create python wrapper to run noise estimation script for named binfhe param sets; add num_inputs/method/num_iterations to inputs; add noise stdev and probability of failure to output
- create python wrapper to generate predicted failure rate and stdev from a set of noise estimation script inputs
- update the main lattice estimator python script to use the default of -40 for probability of failure and 3 digits for key switching
- fix boolean_noise_estimate_script.cpp to use double for stdev

### PR DESCRIPTION
**Pull request by pascoec**
Implements the fixes described in https://github.com/openfheorg/openfhe-development/issues/779

- fixed boolean_noise_estimate_script.cpp to use double instead of int for stdev
- fixed boolean_noise_estimate_script.cpp to calculate noise based on number of gate inputs rather than hardcoded to 3
- updated binfhe_params.py to use the defaults of -40 for probability of failure and 3 digits for key switching
- added upper and lower bound parameters for digit decomposition search space to binfhe_params.py
- added ability to bypass binfhe_params.py parameter selector prompts with command line arguments
- added --all flag functionality to binfhe_params.py
- created binfhe_params_validator.py
- updated README.md
- removed .pyc files
- added .gitignore

### CODE REVIEW

**Comment by yspolyakov** [comment]
* Please remove the *.pyc files
* Update README.md to show the inputs and outputs for the boolean cpp version in the PR

File: scripts/paramsestimator/binfhe_params_helper.py [hunk 1]
```diff
@@ -7,11 +7,10 @@
 import sys
 import os
 import io
-from time import sleep
-#import paramstable as stdparams
+#from time import sleep
 import paramstable as stdparams
 
-syspath = "/home/sara/lattice-estimator"
+syspath = "/home/cpascoe/ops5g/lattice-estimator"
```
Comment (yspolyakov) [inline scripts/paramsestimator/binfhe_params_helper.py]: let us make this more generic and add a comment explaining what this path is used for

**Comment by pascoec** [comment]
> * Please remove the *.pyc files

I think there is some confusion. This PR is already removing pyc files that were previously included [here](https://github.com/openfheorg/openfhe-lattice-estimator/commit/64e85f01c3cb38f71af649117442fa6c85de5b69). Are there others that I missed?

I suggest we add *.pyc to .gitignore

File: scripts/paramsestimator/binfhe_params_helper.py (hunk 1 above)
Comment (pascoec) [reply scripts/paramsestimator/binfhe_params_helper.py]: I suggest we remove this altogether and instruct users to add the lattice-estimator path to PYTHONPATH

**Comment by yspolyakov** [comment]
@pascoec You are correct. I did not notice those files had been previously added. Sure, these need to be removed. 

Let us also add *.pyc to the gitingore list.

File: scripts/paramsestimator/binfhe_params_helper.py (hunk 1 above)
Comment (yspolyakov) [reply scripts/paramsestimator/binfhe_params_helper.py]: Agreed

**Review by yspolyakov — APPROVED** [review]

**Review by yspolyakov — APPROVED** [review]

**Pull request merged**